lib/pbio/src/image/image.c

```diff
@@ -262,6 +262,142 @@ void pbio_image_draw_vline(pbio_image_t *image, int x, int y, int l,
     }
 }
 
+/**
+ * Draw a line with a flat slope (less or equal to 1).
+ * @param [in,out] image      Image to draw into.
```

Comment:
```suggestion
 * @param [in] image      Image to draw into.
```

`[in,out]` would require `pbio_image_t **image` and would return a different pointer than the one that was passed in.

Reply:
I do not agree, the image is read and written, so it’s both in and out.

Random people on the internet seems to agree: https://stackoverflow.com/questions/47732665/is-that-an-in-or-in-out-parameter-doxygen-c :slightly_smiling_face: 

BTW, I find those in/out redundant for C, it’s usually clear from the function prototype.

Reply:
Sure, there is room for interpretation. I learned C# before C, so I've always used the C# semantics for in/out, which are well defined. 

> BTW, I find those in/out redundant for C, it’s usually clear from the function prototype.

This is exactly why I don't consider `[in,out]` correct here. It is only useful for something like:

```c
int read(void* data, size_t *size);
```

Where the precondition is that `*size` is set to the size of `data` in bytes and the post-condition is that `*size` is set to the number of bytes actually read, which may be less than the input size. Just going by the signature alone, one would normally expect `size_t *size` to be an out parameter so `[in,out]` makes it a bit more obvious that something unusual is going on.

Anyway, I'm not going to make a big deal about it. I don't think we've used this 100% consistently throughout the code anyway. But, in the majority of cases, we have only used `[in]` for cases where a struct is modified by a function. So it would be more consistent to stick with just `[in]`.

Reply:
> Sure, there is room for interpretation. I learned C# before C, so I've always used the C# semantics for in/out, which are well defined.

I guess for C# structs though, we would need to call it `[in,out]` like you are doing here since otherwise C# would pass the struct by value. :smile: It's just that normally something like the image type would be an object in C# rather than a struct.

Reply:
OK, made the change for the whole file.